SimCalorimetry/HGCalAssociatorProducers/plugins/SimTauCPLinkProducer.cc

```diff
@@ -0,0 +1,144 @@
+// -*- C++ -*-
+//
+//
+// Authors:  Marco Rovere, Andreas Gruber
+//         Created:  Mon, 16 Oct 2023 14:24:35 GMT
+//
+//
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "DataFormats/HGCalReco/interface/TICLCandidate.h"
+#include "SimDataFormats/CaloAnalysis/interface/CaloParticleFwd.h"
+#include "SimDataFormats/CaloAnalysis/interface/CaloParticle.h"
+#include "SimDataFormats/CaloAnalysis/interface/SimCluster.h"
+#include "SimDataFormats/CaloAnalysis/interface/SimTauCPLink.h"
+
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+
+class SimTauProducer : public edm::stream::EDProducer<> {
```

Comment:
This module could be easily made `edm::global::EDProducer<>`.

Reply:
Hi @makortel,
Thanks for the review and your comments!
Do you think these are urgent changes and should be included in this PR, or will it be sufficient if i address them in a small PR in the near future (given that this one is fully signed already)?

Reply:
The most important comments are
* class version (https://github.com/cms-sw/cmssw/pull/43468#discussion_r1446412230)
* declaring the use of `TFileService` as a shared resource (https://github.com/cms-sw/cmssw/pull/43468#discussion_r1446415528)

Ideally they would be fixed already in this PR (and if done, the `SimTauCPLink_cfi.py` could be removed too, preferably from the commit history altogether). I understand the pain in collecting signatures, although nowadays only those signatures whose packages were changed are reset (so in that sense there is no additional overhead compared to a subsequent PR). It would delay the use of these features by others though.

From `core` perspective I could tolerate merging this PR if a subsequent PR to fix these 2-3 items would be opened ~immediately afterwards (like based on the next IB). The other comments are more for possible future improvements.

@cms-sw/orp-l2 What do you think?